### PR TITLE
Remove old closures compatibility code

### PR DESCRIPTION
**Pull request by stloyd**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| BC breaks?    | no
| License       | MIT

### CODE REVIEW

**Comment by stloyd** [comment]
@pjedrzejewski @michalmarcinkowski @umpirsky ping ;)

File: src/Sylius/Bundle/CoreBundle/Checkout/CheckoutProcessScenario.php [hunk 1]
```diff
@@ -54,21 +48,16 @@ public function build(ProcessBuilderInterface $builder)
...
-        ;
-
-        $builder
             ->setDisplayRoute('sylius_checkout_display')
             ->setForwardRoute('sylius_checkout_forward')
             ->setRedirect('sylius_homepage')
-            ->validate(function () use ($cart) {
-                return !$cart->isEmpty();
+            ->validate(function () {
+                return !$this->cartProvider->getCart()->isEmpty();
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/CoreBundle/Checkout/CheckoutProcessScenario.php]: Why not `$this->getCurrentCart()->isEmpty()`?

**Comment by michalmarcinkowski** [comment]
:+1:

**Comment by umpirsky** [comment]
:+1:

File: src/Sylius/Bundle/CoreBundle/Checkout/CheckoutProcessScenario.php (hunk 1 above)
Comment (stloyd) [inline src/Sylius/Bundle/CoreBundle/Checkout/CheckoutProcessScenario.php]: TBH. I was not sure about this, and decided to not use protected method but direct call, can change if it's a problem.

Comment (umpirsky) [inline src/Sylius/Bundle/CoreBundle/Checkout/CheckoutProcessScenario.php]: +1 for `$this->getCurrentCart()->isEmpty()` for the sake of code readability.

Comment (stloyd) [inline src/Sylius/Bundle/CoreBundle/Checkout/CheckoutProcessScenario.php]: Changed.

**Comment by stloyd** [comment]
@pjedrzejewski ping :)

**Comment by pjedrzejewski** [comment]
Thank you Joseph!

**Pull request merged**